### PR TITLE
refactor(StatusItemSelector): Decompose into smaller components

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Decomposes `StatusItemSelector` into smaller components that can be used as a building blocks in other scenarios:
- `StatusGroupBox` - styled box with a background, title and icon, based on `GroupBox`
- `StatusFlowSelector` - base component for components implementing flow-based selection - implements placeholder item, "add" button and flow layout
- `StatusItemSelector` - based on `StatusFlowSelector` (instead of `Rectangle), the functionality as before.

Closes: https://github.com/status-im/status-desktop/issues/9851

### Affected areas
`StatusItemSelector`

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Screenshot from 2023-03-13 11-38-31
Screenshot from 2023-03-13 11-38-58

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import QtQuick.Layouts 1.15
+
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core 0.1
+
+import utils 1.0
+
+
+GroupBox {
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:12]: It would be fine if you also add a description here!

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import QtQuick 2.15
+
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+/*!
+   \qmltype StatusFlowSelector
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:7]: You need to add it also in `statusqcomponents.qdoc`.

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import QtQuick 2.15
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:1]: You need to add it also in `statusqcomponents.qdoc`.

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    implicitWidth: 560
+    clip: true
+
+    Flow {
+        id: flow
+
+        width: root.availableWidth
+        spacing: 6
+
+        StatusListItemTag {
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:94]: Maybe the placeholder item can change and not being always a `StatusItemListTag`. What about passing the component desired as a parameter?

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            closeButtonVisible: false
+            titleText.color: Theme.palette.baseColor1
+            titleText.font.pixelSize: 15
+        }
+
+        onPositioningComplete: {
+            if (addItemButton.Positioner.isLastItem)
+                return
+
+            addItemButton.parent = null
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:110]: Why is this needed??

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml [hunk 6]
```diff
@@ -1,20 +1,21 @@
...
 import StatusQ.Core.Utils 0.1
 
 /*!
    \qmltype StatusItemSelector
-   \inherits Rectangle
+   \inherits StatusFlowSelector
    \inqmlmodule StatusQ.Components
    \since StatusQ.Components 0.1
-   \brief It allows to add items and display them as a tag item with an image and text. It also allows to store and display logical `and` / `or` operators into the list. Inherits \l{https://doc.qt.io/qt-6/qml-qtquick-rectangle.html}{Item}.
+   \brief It allows to add items and display them as a tag item with an image
```
Comment (noeliaSD) [inline ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:13]: Since I guess your intention for the new airdrop components is create a new component based on `StatusFlowSelector` and not based on `StatusItemSelector`, maybe we can rename that to something more explicit to `tags` mode visualization. WDYT?

File: storybook/pages/StatusItemSelectorPage.qml [hunk 7]
```diff
@@ -55,6 +64,7 @@ ColumnLayout {
 
     StatusButton {
         Layout.alignment: Qt.AlignHCenter
+
```
Comment (noeliaSD) [inline storybook/pages/StatusItemSelectorPage.qml:67]: empty line

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (noeliaSD) [reply ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:94]: Well.. I cannot reproduce that everytime.. sometimes it appears, some times not.. and when it appeared, the permission created looked like this:

Screenshot 2023-03-13 at 14 30 07

**Review by noeliaSD — CHANGES_REQUESTED** [review]
In general LGTM! 
Some minor comments added

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml (hunk 1 above)
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:12]: It's a standard GroupBox I'd say :)

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+       component as a title icon.
+    */
+    property url icon
+
+    /*!
+       \qmlproperty int StatusItemSelector::iconSize
+       This property holds the icon size for the icon represented on top of the
+       component as a title icon.
+    */
+    property int iconSize: 18
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:30]: That looks like a weird icon size; our icons are usually 20 or 24

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml (hunk 4 above)
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:94]: I was thinking about that, but I it doesn't seem needed now. In case of need we can generalize it later imo.

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml (hunk 5 above)
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:110]: bc "add" button needs to the last item in the `Flow`. Without that reordering, `Repeater` defined externally would be the last item and populate it's children also after "add" button.

File: storybook/pages/StatusItemSelectorPage.qml (hunk 7 above)
Comment (micieslak) [reply storybook/pages/StatusItemSelectorPage.qml:67]: hmm, I tend to separate attached properties from the rest, isn't it ok?

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml (hunk 8 above)
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:30]: ok, changed default to 24

**Review by caybro — CHANGES_REQUESTED** [review]
Gotta fix the `imports` test failure too:

```
(qrc:/StatusQ/Components/StatusGroupBox.qml:9:1: module "utils" is not installed
```

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:94]: the first one is fixed, the second cannot reproduce, will take a closer look asap but it seems to be unrelated this change.

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml (hunk 6 above)
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:13]: I didn't invent better name bc we already have `StatusTagsSelector` which is a different thing :)

File: ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+    topPadding: label.implicitHeight
+    leftPadding: 16
+
+    /*!
+       \qmlproperty string StatusGroupBox::icon
+       This property holds the icon name for the icon represented on top of the
+       component as a title icon.
+    */
+    property url icon
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusGroupBox.qml:21]: `string` or `url`? I think `string` is good enough (and more flexible in this case)

File: ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/StatusFlowSelector.qml:94]: Seems like a general problem with the button:

image

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml [hunk 10]
```diff
@@ -93,7 +65,7 @@ Rectangle {
         }
        \endqml
     */
-    property var itemsModel: ListModel { }
+    property var itemsModel
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:68]: Just `model`?

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml [hunk 11]
```diff
@@ -130,116 +102,61 @@ Rectangle {
     */
     signal itemClicked(var item, int index, var mouse)
 
-    color: Theme.palette.baseColor4
-    implicitHeight: columnLayout.implicitHeight + columnLayout.anchors.topMargin + columnLayout.anchors.bottomMargin
+    placeholderItem.visible: repeater.count === 0
+
     implicitWidth: 560
-    radius: 16
     clip: true
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:108]: Need the `clip` here?

**Review by alexjba — APPROVED** [review]
Nice work! 👍

**Review by noeliaSD — APPROVED** [review]

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml (hunk 10 above)
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:68]: yeah, it was `itemsModel` so far but just `model` is better indeed, refactored

File: ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml [hunk 12]
```diff
@@ -130,116 +102,61 @@ Rectangle {
...
-                asset.name: root.defaultItemImageSource
-                asset.isImage: true
+                title: model.text
+
+                asset.height: root.asset.height
+                asset.width: root.asset.width
+                asset.name: root.useLetterIdenticons ? model.text : (model.imageSource ?? "")
+                asset.isImage: root.asset.isImage
+                asset.bgColor: root.asset.bgColor
+                asset.emoji: model.emoji ? model.emoji : ""
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/StatusQ/src/StatusQ/Components/StatusItemSelector.qml:152]: done

**Review by caybro — APPROVED** [review]
LGTM
